The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/27657958807/job/81796194743) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/81796194743) [(plain)](https://github.com/rust-lang/rust/commit/325f8812acce098571deb8eea5df42b8559e600e/checks/81796194743/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
fmt: checked 6923 files
tidy check
tidy [rustdoc_json (src)]: `rustdoc-json-types` modified, checking format version
tidy: Skipping binary file check, read-only filesystem
tidy [style (tests)]: /checkout/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.rs: missing trailing newline
tidy [style (tests)]: FAIL
removing old virtual environment
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'venv'
creating virtual environment at '/checkout/obj/build/venv' using 'python3.10' and 'virtualenv'
Requirement already satisfied: pip in ./build/venv/lib/python3.10/site-packages (26.1.1)
Collecting pip
---
info: ✓ ES-Check passed! All files are ES10 compatible.
typechecking javascript files
tidy: The following check failed: style (tests)
Bootstrap failed while executing `test src/tools/tidy tidyselftest --extra-checks=py,cpp,js,spellcheck`
Command `/checkout/obj/build/x86_64-unknown-linux-gnu/stage1-tools-bin/rust-tidy --root-path=/checkout --cargo-path=/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo --output-dir=/checkout/obj/build --concurrency=4 --npm-path=/node/bin/yarn --ci=true --extra-checks=py,cpp,js,spellcheck` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1624:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1511:29

--- BACKTRACE vvv
   0: <bootstrap::utils::exec::DeferredCommand>::finish_process
             at /checkout/src/bootstrap/src/utils/exec.rs:939:17
   1: <bootstrap::utils::exec::DeferredCommand>::wait_for_output::<&bootstrap::utils::exec::ExecutionContext>
             at /checkout/src/bootstrap/src/utils/exec.rs:831:21
   2: <bootstrap::utils::exec::ExecutionContext>::run
             at /checkout/src/bootstrap/src/utils/exec.rs:741:45
   3: <bootstrap::utils::exec::BootstrapCommand>::run::<&bootstrap::core::builder::Builder>
             at /checkout/src/bootstrap/src/utils/exec.rs:339:27
   4: <bootstrap::core::build_steps::test::Tidy as bootstrap::core::builder::Step>::run
             at /checkout/src/bootstrap/src/core/build_steps/test.rs:1511:29
   5: <bootstrap::core::builder::Builder>::ensure::<bootstrap::core::build_steps::test::Tidy>
             at /checkout/src/bootstrap/src/core/builder/mod.rs:1596:36
   6: <bootstrap::core::build_steps::test::Tidy as bootstrap::core::builder::Step>::make_run
             at /checkout/src/bootstrap/src/core/build_steps/test.rs:1433:21
   7: <bootstrap::core::builder::StepDescription>::maybe_run
             at /checkout/src/bootstrap/src/core/builder/mod.rs:476:13
   8: bootstrap::core::builder::cli_paths::match_paths_to_steps_and_run
             at /checkout/src/bootstrap/src/core/builder/cli_paths.rs:232:18
   9: <bootstrap::core::builder::Builder>::run_step_descriptions
             at /checkout/src/bootstrap/src/core/builder/mod.rs:1139:9
  10: <bootstrap::core::builder::Builder>::execute_cli
             at /checkout/src/bootstrap/src/core/builder/mod.rs:1118:14
  11: <bootstrap::Build>::build
             at /checkout/src/bootstrap/src/lib.rs:803:25
  12: bootstrap::main
             at /checkout/src/bootstrap/src/bin/main.rs:130:11
  13: <fn() as core::ops::function::FnOnce<()>>::call_once
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/core/src/ops/function.rs:250:5
  14: std::sys::backtrace::__rust_begin_short_backtrace::<fn(), ()>
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/std/src/sys/backtrace.rs:166:18
  15: std::rt::lang_start::<()>::{closure#0}
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/std/src/rt.rs:206:18
  16: <&dyn core::ops::function::Fn<(), Output = i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe as core::ops::function::FnOnce<()>>::call_once
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/core/src/ops/function.rs:287:21
  17: std::panicking::catch_unwind::do_call::<&dyn core::ops::function::Fn<(), Output = i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe, i32>
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/std/src/panicking.rs:581:40
  18: std::panicking::catch_unwind::<i32, &dyn core::ops::function::Fn<(), Output = i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe>
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/std/src/panicking.rs:544:19
  19: std::panic::catch_unwind::<&dyn core::ops::function::Fn<(), Output = i32> + core::marker::Sync + core::panic::unwind_safe::RefUnwindSafe, i32>
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/std/src/panic.rs:359:14
  20: std::rt::lang_start_internal::{closure#0}
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/std/src/rt.rs:175:24
  21: std::panicking::catch_unwind::do_call::<std::rt::lang_start_internal::{closure#0}, isize>
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library/std/src/panicking.rs:581:40
---
  28: __libc_start_main
  29: _start


Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:02:40
  local time: Wed Jun 17 00:50:32 UTC 2026
  network time: Wed, 17 Jun 2026 00:50:32 GMT
##[error]Process completed with exit code 1.
##[group]Run echo "disk usage:"
```

</details>

--- a/compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs
+++ b/compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs
@@ -2593,10 +2593,37 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
 
         for init in inits {
             if !variant_def.fields.iter().any(|field_def| field_def.name == init.field.name) {
-                let err = tcx.dcx().struct_span_err(
-                    init.field.span,
-                    format!("struct `{}` has no field named `{}`", variant_def.name, init.field),
-                );
+                let mut err = if adt_def.is_enum() {
+                    struct_span_code_err!(
+                        tcx.dcx(),
+                        init.field.span,
+                        E0559,
+                        "variant `{}::{}` has no field named `{}`",
+                        ty,
+                        variant_def.name,
+                        init.field
+                    )
+                } else {
+                    struct_span_code_err!(
+                        tcx.dcx(),
+                        init.field.span,
+                        E0560,
+                        "struct `{}` has no field named `{}`",
+                        variant_def.name,
+                        init.field
+                    )
+                };
+                if adt_def.is_enum() {
+                    err.span_label(
+                        init.field.span,
+                        format!("`{}::{}` does not have this field", ty, variant_def.name),
+                    );
+                } else {
+                    err.span_label(
+                        init.field.span,
+                        format!("`{}` does not have this field", variant_def.name),
+                    );
+                }
                 return ty::Const::new_error(tcx, err.emit());
             }
         }

--- a/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.rs
+++ b/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.rs
@@ -8,6 +8,6 @@ enum E {
 fn foo<const N: E>() {}
 
 fn main() {
-    E::S { x: const { 1 } };
+    foo::<{E::S { x: const { 1 } }}>();
         //~^ ERROR variant `E::S` has no field named `x` [E0559]
 }

--- a/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.rs
+++ b/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.rs
@@ -1,0 +1,13 @@
+#![feature(min_generic_const_args, adt_const_params)]
+
+#[derive(Eq, PartialEq, std::marker::ConstParamTy)]
+enum E {
+    S {}
+}
+
+fn foo<const N: E>() {}
+
+fn main() {
+    E::S { x: const { 1 } };
+        //~^ ERROR variant `E::S` has no field named `x` [E0559]
+}

--- a/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.stderr
+++ b/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.stderr
@@ -1,10 +1,8 @@
 error[E0559]: variant `E::S` has no field named `x`
-  --> $DIR/adt_expr_unit_enum_extra_field.rs:11:12
+  --> $DIR/adt_expr_unit_enum_extra_field.rs:11:19
    |
-LL |     E::S { x: const { 1 } };
-   |            ^ `E::S` does not have this field
-   |
-   = note: all struct fields are already assigned
+LL |     foo::<{E::S { x: const { 1 } }}>();
+   |                   ^ `E::S` does not have this field
 
 error: aborting due to 1 previous error
 

--- a/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.stderr
+++ b/tests/ui/const-generics/mgca/adt_expr_unit_enum_extra_field.stderr
@@ -1,0 +1,11 @@
+error[E0559]: variant `E::S` has no field named `x`
+  --> $DIR/adt_expr_unit_enum_extra_field.rs:11:12
+   |
+LL |     E::S { x: const { 1 } };
+   |            ^ `E::S` does not have this field
+   |
+   = note: all struct fields are already assigned
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0559`.

--- a/tests/ui/const-generics/mgca/adt_expr_unit_struct_extra_field.rs
+++ b/tests/ui/const-generics/mgca/adt_expr_unit_struct_extra_field.rs
@@ -7,5 +7,5 @@ fn foo<const N: Foo>() {}
 
 fn main() {
     foo::<{ Foo { field: const { 1 } } }>();
-    //~^ ERROR struct `Foo` has no field named `field`
+    //~^ ERROR struct `Foo` has no field named `field` [E0560]
 }

--- a/tests/ui/const-generics/mgca/adt_expr_unit_struct_extra_field.stderr
+++ b/tests/ui/const-generics/mgca/adt_expr_unit_struct_extra_field.stderr
@@ -1,8 +1,9 @@
-error: struct `Foo` has no field named `field`
+error[E0560]: struct `Foo` has no field named `field`
   --> $DIR/adt_expr_unit_struct_extra_field.rs:9:19
    |
 LL |     foo::<{ Foo { field: const { 1 } } }>();
-   |                   ^^^^^
+   |                   ^^^^^ `Foo` does not have this field
 
 error: aborting due to 1 previous error
 
+For more information about this error, try `rustc --explain E0560`.

